.github/workflows/test-pr.yml

```diff
@@ -0,0 +1,74 @@
+name: Test PR on Linux and macOS
+
+on:
+  push:
+    branches:
+      - pixi
+
+jobs:
+  test:
+    name: Test on ${{ matrix.os }}
+    runs-on: ${{ matrix.os }}
+    timeout-minutes: 30
+    strategy:
+      matrix:
+        os: [ubuntu-latest, macos-latest]
+      fail-fast: false
+
+    steps:
+      # Checkout the repository
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      # Install Pixi
+      - name: Install Pixi
+        uses: prefix-dev/setup-pixi@v0.8.10
+        with:
+          pixi-version: v0.25.0
+          cache: false
+          auth-host: prefix.dev
+          auth-token: ${{ secrets.PREFIX_DEV_TOKEN }}
```

Comment:
Just a quick comment: I don't think we need any authentication here – we don't want to push anything, and we download only from public channels.